### PR TITLE
Add the ability for technician and rocket soldier to fire from a pillbox

### PR DESCRIPTION
**Pull request by PunkPun**
https://github.com/OpenRA/OpenRA/pull/18570#issuecomment-739564660

### CODE REVIEW

**Comment by AspectInteractive2** [comment]
There seems to be two ranged circles when the rocket soldier enters the pillbox, yet only the inner ranged circle is valid for the rocket soldier (it cannot shoot as far as the outer range circle).

image

**Comment by PunkPun** [comment]
the second one is for anti-air

**Comment by AspectInteractive2** [comment]
Can it be a different colour? I can easily see people being confused by why there are two circles, and which one is for ground vs. which one is for air.

**Comment by PunkPun** [comment]
>  I can easily see people being confused by why there are two circles, and which one is for ground vs. which one is for air.

Which is fine. It's just another way to hint at people that rocket soldier has different ranges for ground and for air

**Comment by anvilvapre** [comment]
what visually might work is to have a larger spacing between the dashes of the outer circle. to emphasize they are two different types of ranges.

**Comment by PunkPun** [comment]
I'm not sure how that would make it clearer than it currently is

File: mods/ra/rules/infantry.yaml
Comment (Mailaender) [inline mods/ra/rules/infantry.yaml:199]: This is a balance change. The anti-ground bunker suddenly becomes AA.

Comment (PunkPun) [reply mods/ra/rules/infantry.yaml:199]: It is a balance change, but an incredibly insignificant one at that. A single rocket soldier does little damage to aircraft and a rocket pillbox is as expensive the overpowered AA gun. 800$ vs 800$ (600 + 300 - 100)

I've also seen it tested in quite a lot of older balance maps, its fairly weak without additional buffs

Comment (PunkPun) [reply mods/ra/rules/infantry.yaml:199]: I've also seen it tested in quite a lot of older balance maps, its fairly weak without additional buffs

Comment (ssyrnyk) [reply mods/ra/rules/infantry.yaml:199]: I would like to comment that this seems like a good idea. It makes the game more consistent logically and the AA unbalancing would be a minor effect if at all.

**Review by RoosterDragon — APPROVED** [review]

**Review by tttppp — APPROVED** [review]
This looks good to me. In Discord you mentioned medics and mechanics acting from PBs was considered immersion breaking, but I'd be interested to see these have garrisoned 'attacks' too.

**Review by penev92 — APPROVED** [review]
*This needs at least 1 approving review by someone with write access.*

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/767521a6a65858c013928e796dca635e26521e6a...1d18391f463a2f601edb005b9ba455b4930273bf)

**Pull request merged**
